lib/grpcApi/handlers/commitTransactionHandlerFactory.js

```diff
@@ -0,0 +1,64 @@
+const { CommitTransactionResponse } = require('@dashevo/drive-grpc');
+
+const InvalidArgumentGrpcError = require('../error/InvalidArgumentGrpcError');
+const InternalGrpcError = require('../error/InternalGrpcError');
+
+/**
+ * @param {MongoDBTransaction} stateViewTransaction
+ * @param {SVContractMongoDbRepository} svContractMongoDbRepository
+ * @param {createContractDatabase} createContractDatabase
+ * @param {removeContractDatabase} removeContractDatabase
+ * @returns {commitTransactionHandler}
+ */
+module.exports = function commitTransactionHandlerFactory(
+  stateViewTransaction,
+  svContractMongoDbRepository,
+  createContractDatabase,
+  removeContractDatabase,
+) {
+  /**
+   * Commit transaction, that was created before and create collections for
+   * documents inside of received contracts
+   *
+   * @typedef commitTransactionHandler
+   * @param {Object} call
+   * @returns {Promise<CommitTransactionResponse>}
+   */
+  async function commitTransactionHandler(call) {
+    const { request } = call;
+
+    const blockHash = request.getBlockHash();
+
+    if (!blockHash) {
+      throw new InvalidArgumentGrpcError('blockHash is not specified');
+    }
+
+    const contracts = await svContractMongoDbRepository
+      .findAllByReferenceSTHash(blockHash, stateViewTransaction);
+
+    const createdContracts = [];
+
+    try {
+      for (const contract of contracts) {
+        await createContractDatabase(contract);
+        createdContracts.push(contract);
+      }
+
+      await stateViewTransaction.commit();
```

Comment:
If we commit first and then create contract databases then logic will be much simpler.

Reply:
Everyone has a different opinion on this. Let's discuss about it.

Reply:
If we commit and then create databases, how do we roll back the transaction if database creation failed?

Reply:
Yeah, you right!